python/dgl/transform.py

```diff
@@ -645,4 +788,68 @@ def out_subgraph(g, nodes):
         ret.edges[etype].data[EID] = induced_edges[i].tousertensor()
     return ret
 
+def to_simple(g, return_counts=None, writeback_mapping=False):
+    """Convert a heterogeneous multigraph to a heterogeneous simple graph, coalescing
+    duplicate edges into one.
+
+    This function does not preserve node and edge features.
+
+    Parameters
+    ----------
+    g : DGLHeteroGraph
+        The heterogeneous graph
+    return_counts : str, optional
+        If given, the returned graph would have a column with the same name that stores
+        the number of duplicated edges from the original graph.
+    writeback_mapping : bool, default False
+        If True, the mapping from the edge IDs of original graph to those of the returned
+        graph would be written into edge feature ``dgl.EID`` in the original graph for
+        each edge type.
```

Comment:
Don't like this write-back behavior. Consider return the mapping optionally.

Reply:
It will return a dictionary of mappings by edge type though.